client/services/FeedbackGeneratorService.js

```diff
@@ -129,13 +132,15 @@ tie.factory('FeedbackGeneratorService', [
      *    name for the function needed to process/format the user's output.
      * @param {CorrectnessTest} correctnessTest
      * @param {*} observedOutput Actual output for running user's code.
+     * @param {Feedback} oldFeedback Feedback object to append feedback to. If
```

Comment:
Yeah that makes sense. I've added the appendFeedback helper method to the `Feedback` object class now and have used that instead in my implementation. It has definitely helped with simplifying much of the code.